### PR TITLE
App View header

### PR DESCRIPTION
**Pull request by hetunandu**

### CODE REVIEW

File: app/client/src/constants/messages.ts [hunk 1]
```diff
@@ -136,7 +136,7 @@ export const AUTOFIT_COLUMN = "Autofit column";
 export const TIMEZONE = "Timezone";
 export const ENABLE_TIME = "Enable Time";
 
-export const EDIT_APP = "Edit App";
+export const EDIT_APP = "Back to editor";
```
Comment (Nikhil-Nandagopal) [inline app/client/src/constants/messages.ts:139]: This should be edit itself because user can land on launch view from share / home page and other channels

File: app/client/src/components/editorComponents/Button.tsx [hunk 2]
```diff
@@ -86,7 +87,7 @@ export type ButtonProps = {
   text?: string;
   onClick?: () => void;
   href?: string;
-  icon?: string;
+  icon?: string | MaybeElement;
```
Comment (Nikhil-Nandagopal) [inline app/client/src/components/editorComponents/Button.tsx:90]: why is this needed?

File: app/client/src/constants/messages.ts (hunk 1 above)
Comment (hetunandu) [reply app/client/src/constants/messages.ts:139]: Following the designs here https://www.figma.com/file/dAlfdCqJUxiNfJIBeQ1cuq/appsmith_widgetZ?node-id=1156%3A528

File: app/client/src/components/editorComponents/Button.tsx (hunk 2 above)
Comment (hetunandu) [reply app/client/src/components/editorComponents/Button.tsx:90]: To pass a custom icon component inside a button. Blueprint supports it but it was not allowed in these props.
